manifests/resource/location.pp

```diff
@@ -153,66 +153,65 @@
 #  }
 
 define nginx::resource::location (
-  $ensure                      = present,
-  $internal                    = false,
-  $location                    = $name,
-  $server                      = undef,
-  $www_root                    = undef,
-  $autoindex                   = undef,
-  $index_files                 = [
+  Enum['present', 'absent'] $ensure                   = present,
+  Boolean $internal                                   = false,
+  String $location                                    = $name,
+  String $server                                      = undef,
```

Comment:
Yeah, old code does look a bit funky.  Do you want to remove the default then since now it is much more obvious it provides no value.